src/screens/MapPage/MapPage.js

```diff
@@ -23,6 +28,27 @@ export default function MapPage() {
           <MapDetail
             coordinates={miniMapCoords}
             onClose={() => setCoords(null)}
+            setActiveSubject={setActiveSubject}
+            setShowSubjectsModal={setShowSubjectsModal}
+          />
+        </Layer>
+      )}
+
+      {activeSubject && (
+        <Layer>
+          <MetadataModal
+            onClose={setActiveSubject}
+            subject={activeSubject}
+          />
+        </Layer>
+      )}
+
+      {showSubjectsModal && (
+        <Layer>
```

Comment:
I'd add the `onEsc` prop to this so that key also closes the modal. 
```suggestion
        <Layer
          onEsc={() => setShowSubjectsModal(false)}
        >
```